### PR TITLE
Holidaysdifference

### PR DESCRIPTION
**Pull request by shibe2018**

### CODE REVIEW

File: modules/cbMap/processmap/InformationMap.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * granted by the License. coreBOS distributed by JPL TSolucio S.L. is distributed in
+ * the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. Unless required by
+ * applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT ANY WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License. You may obtain a copy of the License
+ * at <http://corebos.org/documentation/doku.php?id=en:devel:vpl11>
+ *************************************************************************************************
+ *  Module       : Business Mappings:: Module to Module Mapping
```
Comment (joebordes) [inline modules/cbMap/processmap/InformationMap.php:16]: Generic Information Map

File: modules/cbMap/processmap/InformationMap.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ *  Version      : 1.0
+ *  Author       : JPL TSolucio, S. L.
+ *************************************************************************************************
+ * The accepted format is:
+<map>
+	<information>
+		<infotype>Italy Holidays</infotype>
```
Comment (joebordes) [inline modules/cbMap/processmap/InformationMap.php:23]: Just "Holidays", it should be the "type" or "category" of the information, the specifics will be in the map name

Comment (shibe2018) [reply modules/cbMap/processmap/InformationMap.php:23]: so i should remove **Italy**

File: modules/com_vtiger_workflow/expression_engine/VTExpressionsManager.inc [hunk 3]
```diff
@@ -152,6 +152,7 @@ class VTExpressionsManager {
 			'allrelatedare' => 'allrelatedare(relatedmodule, fieldname, value)',
 			'min' => 'min(fieldname)',
 			'max' => 'max(fieldname)',
+			'holidaydifference' => 'holidaydifference(startDate, endDate,mapname, include_saturdays)',
```
Comment (joebordes) [inline modules/com_vtiger_workflow/expression_engine/VTExpressionsManager.inc:155]: change this:
- the parameters should be: (startDate, endDate, include_saturdays, mapname)  to be the same as the other holiday functions
- move the whole line to be next to the other holiday functions

Comment (shibe2018) [reply modules/com_vtiger_workflow/expression_engine/VTExpressionsManager.inc:155]: okay understood

File: modules/com_vtiger_workflow/expression_functions/datetime.php [hunk 4]
```diff
@@ -42,6 +42,58 @@ function __vt_time_diff($arr) {
...
+	while ($firstDate->diff($lastDate)->days > 0) {
+		$days += $firstDate->format('N') < $lastdow ? 1 : 0;
+		$firstDate = $firstDate->add($oneDay);
+	}
+
+	$cbMapid = GlobalVariable::getVariable('BusinessMapping_'.$mapname, cbMap::getMapIdByName($mapname));
+	if ($cbMapid) {
+		$cbMap = cbMap::getMapByID($cbMapid);
+		$holidays = $cbMap->InformationMap()->readInformationValue();
+		print_r('dates         '.$holidays);
```
Comment (joebordes) [inline modules/com_vtiger_workflow/expression_functions/datetime.php:85]: eliminate this

**Comment by joebordes** [comment]
thanks!

**Pull request merged**